ccloader/js/mod.js

```diff
@@ -118,7 +128,57 @@ export class Mod {
 	get baseDirectory(){
 		return this._getBaseName(this.file).replace(/\\/g, '/').replace(/\/\//g, '/') + '/';
 	}
-	
+
+	/**
+	 * Dynamically add file paths by extensions.
+	 * @param {string[]} fileExtensions
+	 * @param {string} [relativePath = ''] to limit search in
+	 * @returns {Promise<boolean>} true if mod is ready and mod is running, otherwise false
+	 */
+	async addAssetsByExtensions(fileExtensions, relativePath = '') {
+		if (!this.ready) {
+			return false;
+		}
+
+		if (!(window.isLocal || this.packed)) {
+			return false;
+		}
+
+		await this._addFilesToAssets(relativePath, fileExtensions);
+		
+		return true;
+	}
+
+
+	/**
+	 *  Adds all files in folder to path
+	 * @param {string} relativePath to folder
+	 * @returns {Promise<boolean>} true if mod is ready and mod is running, otherwise false
+	 */
+	async addAssetsInFolder(relativePath) {
```

Comment:
It would make sense to merge `addAssetsByExtensions` and `addAssetsInFolder` into a single function `addAssets`.